Bindings/Java/Matlab/Utilities/osimC3D.m

```diff
@@ -0,0 +1,127 @@
+classdef osimC3D < matlab.mixin.SetGet
+% osimC3D C3D data into OpenSim Tables
+%   Utility  Class that uses OpenSim's C3DFileAdapter() class to produce
+%   OpenSim tables. Methods include Simbody rotation.
+    properties (Access = private)
+        path
+        markers
+        forces
+    end
+    methods
+        function obj = osimC3D(path2c3d)
+            % Class Constructor: input is an absolute path to a C3D file. 
+            
+            % verify the file path is correct
+            if exist(path2c3d, 'file') == 0
+                error('File does not exist. Check path is correct')
+            else
+                obj.path = path2c3d;
+            end
+            % load java libs
+            import org.opensim.modeling.*
+            % Use a c3dAdapter to turn read a c3d file
+            tables = C3DFileAdapter().read(path2c3d);
+            % get the marker and force data into OpenSim tables
+            obj.markers = tables.get('markers');
+            obj.forces = tables.get('forces');
+        end            
+        function rate = getRate_marker(obj)
+            % Get the capture rate used for the Marker Data
+            rate = str2double(obj.markers.getTableMetaDataAsString('DataRate'));
+        end
+        function rate = getRate_force(obj)
+            % Get the capture rate used for the Force Data
+            rate = str2double(obj.forces.getTableMetaDataAsString('DataRate'));
+        end
+        function n = getNumTrajectories(obj)
+            % Get the number of markers in the c3d file
+            n = obj.markers.getNumColumns();
+        end
+        function n = getNumForces(obj)
+            % Get the number of forceplates in the c3d file
+            n = (obj.forces.getNumColumns())/3;
+        end
+        function t = getStartTime(obj)
+            % Get the start time of the c3d file
+            t = obj.markers.getIndependentColumn().get(0);
+        end
+        function t = getEndTime(obj)
+            % Get the end time of the c3d file
+            t = obj.markers.getIndependentColumn().get(obj.markers.getNumRows() - 1);
+        end
+        function name = getFileName(obj)
+            % Get the name of the c3d file
+            [filedirectory, name, extension] = fileparts(obj.path);
+        end
+        function filedirectory = getDirectory(obj)
+            % Get the directory path for the c3d file
+            [filedirectory, name, extension] = fileparts(obj.path);
+        end
+        function table = getTable_markers(obj)
+            table = obj.markers().clone();
+        end
+        function table = getTable_forces(obj)
+            table = obj.forces().clone();
+        end
+        function [markerStruct, forcesStruct] = getAsStructs(obj)
```

Comment:
Is this still necessary, if we add the ability to transform tables of vector data to the API? Or is this the primary to get data out for plotting, etc... in MATLAB?

Reply:
I added this method as a convenience for users. So no instead of this;
```
marker_table = osimC3D().getTable_markers();
marker_struct = osimTableToStruct(marker_table);
force_table = osimC3D().getTable_forces();
force_struct = osimTableToStruct(force_table);
```
you can just do this;
```
[marker_struct, forces_struct] = osimC3D.getAsStructs()
```
That was the only reason.

Reply:
Makes sense. Thanks for clarifying.